msteams/provider/provider.py

```diff
@@ -0,0 +1,89 @@
+import logging
+
+from bs4 import BeautifulSoup
+from .unstructured import get_unstructured_client
+from .client import get_client
+
+logger = logging.getLogger(__name__)
+
+cached_pages = {}
+token = None
+
+
+def strip_html_tags(html_text):
+    soup = BeautifulSoup(html_text, "html.parser")
+    return soup.get_text()
+
+
+def serialize_results(results):
+    return [
+        {
+            "id": str(result["id"]),
+            "title": str(
+                result["subject"]
+                or strip_html_tags(result["body"]["content"])[:30] + "..."
+            ),
+            "text": strip_html_tags(result["body"]["content"]),
+            "summary": str(result["summary"]),
+            "url": str(
+                result["link"]
+                if "link" in result
+                else result["webUrl"]
+                if "webUrl" in result
+                else ""
+            ),
+            "has_attachments": str(len(result["attachments"]) > 0),
+            "date": str(result["createdDateTime"]),
+        }
+        for result in results
+    ]
+
+
+def parse_results_attachments(results):
+    attachments_to_unstructured = []
+    for result in results:
+        attachments = result["attachments"]
+        if attachments:
+            for attachment in attachments:
```

Comment:
`attachments_to_unstructured.extend([attachment for attachment in attachments if attachment["contentType"] == "reference"])`

Reply:
In general, let's prevent heavily nested blocks of code